src/main/java/seedu/address/model/person/Person.java

```diff
@@ -190,16 +190,20 @@ public HashMap<AbsentDate, AbsentReason> getAttendances() {
     }
 
     /**
-     * Returns true if both persons have the same name.
+     * Returns true if both persons have the same register number and class, or same phone numbers, or same emails.
      * This defines a weaker notion of equality between two persons.
      */
     public boolean isSamePerson(Person otherPerson) {
-        if (otherPerson == this) {
-            return true;
+        if (otherPerson == null) {
+            return false;
         }
 
-        return otherPerson != null
-                && otherPerson.getName().equals(getName());
+        boolean sameClass = (otherPerson.getStudentClass().equals(this.getStudentClass()));
+        boolean sameRegNo = (otherPerson.getRegisterNumber().equals(this.getRegisterNumber()));
+        boolean samePhone = (otherPerson.getPhone().equals(this.getPhone()));
+        boolean sameEmail = (otherPerson.getEmail().equals(this.getEmail()));
+
+        return sameClass && sameRegNo || samePhone || sameEmail;
```

Comment:
```suggestion
        return (sameClass && sameRegNo) || samePhone || sameEmail;
```
not too sure if returns the same result, but just to be sure

Reply:
okay yup makes sense to be safer